src/core/mavsdk_impl.cpp

```diff
@@ -239,11 +232,27 @@ bool MavsdkImpl::send_message(mavlink_message_t& message)
 {
     std::lock_guard<std::mutex> lock(_connections_mutex);
 
+    uint8_t successful_emissions = 0;
     for (auto it = _connections.begin(); it != _connections.end(); ++it) {
-        if (!(**it).send_message(message)) {
-            LogErr() << "send fail";
-            return false;
+        // Checks whether connection knows target system ID by extracting target system if set.
+        // https://github.com/PX4/PX4-Autopilot/blob/v1.11.3/src/modules/mavlink/mavlink_main.cpp#L472
+        const mavlink_msg_entry_t* meta = mavlink_get_msg_entry(message.msgid);
+
+        if (meta && meta->flags & MAV_MSG_ENTRY_FLAG_HAVE_TARGET_SYSTEM) {
+            int target_system_id = (_MAV_PAYLOAD(&message))[meta->target_system_ofs];
+            if (!(**it).has_system_id(target_system_id)) {
+                continue;
+            }
         }
+
+        if ((**it).send_message(message)) {
+            successful_emissions++;
+        }
+    }
+
+    if (successful_emissions == 0) {
```

Comment:
That's an interesting one :smile:. Is a message successfully sent if it was sent on at least one connection, or if it was sent on all connections? Here we decide that it is successfully sent if it could be sent on one connection, correct?

Reply:
Yes, but as you said it depends on how we want to interpret a successful emission.
But for me, the main thing was to let the loop iterate over all connections even if one fails to send the message.

Reply:
ok :+1: